feat!: use autodoc inventory for automatic typegen

teaishealthy opened the pull request:
Can be merged when next ⟶ stable on [eludris/eludris](https://github.com/eludris/eludris)

Using the autodoc inventory for typegen means:
- Easier and faster type updates
- Full docs coverage
- !: We can't split types into multiple files easily

teaishealthy committed:
feat!: use autodoc inventory for automatic typegen

Using the autodoc inventory for typegen means:
- Easier and faster type updates
- Full docs coverage
-! We can't split types into multiple files easily

spifory reviewed (changes requested):
Now that this uses v0.3.3, maybe the docs link in the README file should be changed to https://eludevs.pages.dev

https://github.com/eludris-community/eludris-api-types.ts/blob/cdacc95d205816fa16b7445925ede3671870eb9d/README.md?plain=1#L7

teaishealthy committed:
feat(autotype): automatic version detection

teaishealthy committed:
chore: update types

teaishealthy committed:
fix: make types actually importable

teaishealthy committed:
feat: add v0.4.0-alpha1

This will be removed once real v0.4 is here

teaishealthy committed:
fix: dont include extension

teaishealthy committed:
fix: prefix enum variants with their name

teaishealthy committed:
fix: no wrong flatten

fix: bases for structs

teaishealthy committed:
docs: add docs

teaishealthy committed:
feat: use a real code-generation framework

This reduces bugs, yields much better doc comment, reduces overall code size
We also get routes

teaishealthy committed:
docs: fix typedoc.json

teaishealthy committed:
chore: add docs command

teaishealthy committed:
docs: update readme
- add link to docs

teaishealthy committed:
chore: remove tinker.ts

teaishealthy committed:
chore: bump

teaishealthy committed:
fix: bugs

teaishealthy committed:
docs: add api.json inventory

teaishealthy committed:
docs(README): change Discord to Eludris (#2) (#3)

Co-authored-by: toolifelesstocode <luvuyosilubane50@gmail.com>
